src/types/timestamp.rs

```diff
@@ -1584,6 +1644,114 @@ mod timestamp_tests {
         Ok(())
     }
 
+    #[test]
+    fn test_timestamp_year() -> IonResult<()> {
+        let timestamp = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp.year(), 2021);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_month() -> IonResult<()> {
+        let timestamp = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp.month(), 2);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_day() -> IonResult<()> {
+        let timestamp_1 = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp_1.day(), 1);
+
+        let timestamp_2 = Timestamp::with_year(2021)
+            .with_month(2)
+            .with_day(4)
+            .build()?;
+
+        assert_eq!(timestamp_2.day(), 4);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_hour() -> IonResult<()> {
```

Comment:
Could you add a unit test showing that `hour()` works as expected when the offset is a `15` or `30` minute interval? (e.g. Chennai)